lib/cache/kube.go

```diff
@@ -235,22 +234,13 @@ func newKubernetesWaitingContainerCollection(upstream services.KubeWaitingContai
 			proto.CloneOf[*kubewaitingcontainerv1.KubernetesWaitingContainer],
 			map[kubeWaitingContainerIndex]func(*kubewaitingcontainerv1.KubernetesWaitingContainer) string{
 				kubeWaitingContainerNameIndex: func(u *kubewaitingcontainerv1.KubernetesWaitingContainer) string {
-					return u.GetMetadata().GetName()
+					return u.GetSpec().GetUsername() + "/" + u.GetSpec().GetCluster() + "/" + u.GetSpec().GetNamespace() + "/" + u.GetSpec().GetPodName() + "/" + u.GetMetadata().GetName()
```

Comment:
This same construction is done in two other places - `ListKubernetesWaitingContainers()` which is identical and `GetKubernetesWaitingContainer()` which constructs the cache key from a request and not a resource.

At the very least, can this be factored out into a separate function that is used here and in `ListKubernetesWaitingContainers()`? It would be nice if it could also be used by `GetKubernetesWaitingContainer()` but I don't immediately see how.

This would help prevent them diverging in future causing this issue again.